lib/deppack.js

```diff
@@ -0,0 +1,289 @@
+  'punycode': {},
+  'querystring': {},
+  'readline': {},
+  'repl': {},
+  'string_decoder': {},
+  'tls': {},
+  'tty': {},
+  'url': {},
+  'util': {},
+  'vm': {},
+  'zlib': {}
+};
+
+const id = x => x;
+const uniq = arr => arr.filter((i, pos) => arr.indexOf(i) === pos);
+const not = f => i => !f(i);
+
+const readFile = (path, callback) => {
+  return fs.readFile(path, {encoding: 'utf8'}, callback);
+};
+
+const getModuleRootPath = path => {
+  const split = path.split('/');
+  const index = split.lastIndexOf('node_modules');
+  return split.slice(0, index + 2).join('/');
+};
+
+const getModuleRootName = path => {
+  const split = path.split('/');
+  const index = split.lastIndexOf('node_modules');
+  return split[index + 1];
+};
+
+const getModuleMainPath = path => sysPath.dirname(getMainFile(path));
+const packageJson = filePath => require(relativeToRoot(filePath, 'package.json'));
+const browserMappings = filePath => packageJson(filePath).browser || {};
+const relativeToRoot = (filePath, relPath) => sysPath.join(getModuleRootPath(filePath), relPath);
+
+const isRelative = path => path.slice(0, 2) === './';
+
+const globalBrowserMappings = filePath => {
+  const brMap = browserMappings(filePath);
+
+  const newBrMap = {};
+
+  Object.keys(brMap).filter(not(isRelative)).forEach(key => {
+    const val = brMap[key];
+    newBrMap[key] = generateModuleName(relativeToRoot(filePath, val));
+  });
+
+  return newBrMap;
+};
+
+const expandedFilePath = filePath => {
+  const brMap = browserMappings(filePath);
+
+  Object.keys(brMap).filter(isRelative).forEach(key => {
+    const val = brMap[key];
+    if (filePath === relativeToRoot(filePath, val)) {
+      filePath = relativeToRoot(filePath, key);
+    }
+  });
+
+  return filePath;
+};
+
+const isMain = filePath => getMainFile(filePath) === filePath;
+
+const getNewHeader = (moduleName, source, filePath) => {
+  const brMap = globalBrowserMappings(filePath);
+
+  const r = isMain(filePath) ? "function(n) { return req(n.replace('./', '" + moduleName + "/')); }" : "req";
+
+  return `require.register('${moduleName}', function(exports,req,module){
+    var require = __makeRequire((${r}), ${JSON.stringify(brMap)});
+    ${source}
+  });`;
+};
+
+const generateModule = (filePath, source) => {
+  filePath = expandedFilePath(filePath);
+  const rootMod = getModuleRootName(filePath);
+  const mn = generateModuleName(filePath);
+
+  return getNewHeader(mn, source, filePath);
+};
+
+const generateModuleName = filePath => {
+  const rp = getModuleMainPath(filePath);
+  const mn = getModuleRootName(filePath) + (isMain(filePath) ? '' : filePath.replace(rp, '').replace('.js', ''));
+
+  return mn;
+};
+
+const _mainFile = (root, json) => {
+  const depMain = typeof json.browser === 'string' ? json.browser : json.main || 'index.js';
+  const fileOrDir = sysPath.join(root, depMain);
+  var isDir;
+  try { isDir = fs.lstatSync(fileOrDir).isDirectory() } catch(e) { isDir = false }
```

Comment:
sync operations is a big "no". Brunch should always use async methods when possible.